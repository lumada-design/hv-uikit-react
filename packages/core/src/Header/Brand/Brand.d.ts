import * as React from "react";
import { StandardProps } from "@mui/material";

export type HvHeaderBrandClassKey = "root" | "separator";

export interface HvHeaderBrandProps
  extends StandardProps<React.HTMLAttributes<HTMLDivElement>, HvHeaderBrandClassKey> {
  /**
   * The brand image node.
   */
  logo?: React.ReactNode;
  /**
   * The brand name string.
   */
  name?: string;
}

export default function HvHeaderBrand(props: HvHeaderBrandProps): JSX.Element | null;
