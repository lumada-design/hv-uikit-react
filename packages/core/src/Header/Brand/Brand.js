/*
 * Copyright 2019 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import PropTypes from "prop-types";
import isNill from "lodash/isNil";
import classNames from "classnames";
import HvTypography from "../../Typography";
import ImageContainer from "../ImageContainer";

/**
 * Separator element between company and product information.
 *
 * @param classes
 * @returns {*}
 * @constructor
 */
// eslint-disable-next-line react/prop-types
const Separator = ({ classes }) => (
  <div className={classes.separatorBox}>
    <div className={classes.separator} />
    <div className={classes.separatorPadding} />
  </div>
);

/**
 * Product text title.
 *
 * @param textContainerClassName
 * @param text
 * @returns {*}
 * @constructor
 */
// eslint-disable-next-line react/prop-types
const ElementContainer = ({ classes, text, existLogo }) => (
  <div
    className={classNames(classes.productLogoTextContainer, {
      [classes.paddingTextLeft]: existLogo
    })}
  >
    <HvTypography variant="highlightText">{text}</HvTypography>
  </div>
);

/**
 * Brand container. It may contain a company logo, product logo and product text.
 *
 * @param classes
 * @param companyLogo
 * @param productLogo
 * @param productText
 * @returns {*}
 * @constructor
 */

/*
*
* className={classnames(
          disabled
            ? classes.disabledLabel
            : {
                [classes.labelSelected]: !clickState,
                [classes.labelDeselected]: clickState
              },
          classes.labelLeftPositioning
        )}
* */
const Brand = ({
  classes,
  centerAlignElement,
  companyLogo,
  productLogo,
  productText
}) => {
  const existSeparator =
    !isNill(companyLogo) && !(isNill(productLogo) && isNill(productText));

  return (
    <div
      className={classNames(classes.brandContainer, {
        [classes.centerBrandContainer]: centerAlignElement
      })}
    >
      {companyLogo && (
        <ImageContainer
          image={companyLogo}
          containerClassName={classes.iconContainer}
        />
      )}
      {existSeparator && <Separator classes={classes} />}
      {productLogo && (
        <ImageContainer
          image={productLogo}
          containerClassName={classes.iconContainer}
        />
      )}
      {productText && (
        <ElementContainer
          classes={classes}
          text={productText}
          existLogo={!isNill(productLogo)}
        />
      )}
    </div>
  );
};

Brand.propTypes = {
  /**
   * A Jss Object used to override or extend the component styles.
   */
  classes: PropTypes.instanceOf(Object).isRequired,
  /**
   * Company logo. Can be a path for a image or a component.
   */
  centerAlignElement: PropTypes.bool,
  /**
   * Company logo. Can be a path for a image or a component.
   */
  companyLogo: PropTypes.node,
  /**
   * Product logo. Can be a path for a image or a component.
   */
  productLogo: PropTypes.node,
  /**
   * Product text.
   */
  productText: PropTypes.string
};

Brand.defaultProps = {
  centerAlignElement:null,
  companyLogo: null,
  productLogo: null,
  productText: null
};

export default Brand;
