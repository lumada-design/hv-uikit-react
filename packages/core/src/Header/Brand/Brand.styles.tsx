import { createClasses } from "@hitachivantara/uikit-react-utils";
import { theme } from "@hitachivantara/uikit-styles";

export const { staticClasses, useClasses } = createClasses("HvHeaderBrand", {
  root: { display: "flex", alignItems: "center" },
  separator: {
    width: 1,
    height: theme.space.sm,
    margin: theme.space.xs,
    backgroundColor: theme.colors.text,
  },
  brandName: {},
});
