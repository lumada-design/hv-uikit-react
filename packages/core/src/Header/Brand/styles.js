/*
 * Copyright 2019 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const styles = theme => ({
  brandContainer: {
    position: "relative",
    display: "inherit",
    alignItems: "center",
    height: "100%",
    marginRight: `${theme.hv.spacing.lg}px`,
    zIndex:2
  },

  centerBrandContainer: {
    flex: 1,
    display: "flex",
    justifyContent: "center"
  },
  iconContainer: {
    display: "inherit",
    maxHeight: "46px"
  },
  companyLogoTextContainer: {
    height: "46px",
    display: "flex",
    alignItems: "center"
  },
  separator: {
    width: `${theme.hv.spacing.xs}px`,
    height: `${theme.hv.spacing.sm}px`,
    borderRight: `1px solid ${theme.hv.palette.accent.acce1}`
  },
  separatorPadding: {
    width: `${theme.hv.spacing.xs}px`
  },
  separatorBox: {
    display: "flex"
  },
  productLogoTextContainer: {
    height: "46px",
    display: "flex",
    alignItems: "center",
    whiteSpace: "nowrap"
  },
  paddingTextLeft: { paddingLeft: `${theme.hv.spacing.xs}px` }
});

export default styles;
