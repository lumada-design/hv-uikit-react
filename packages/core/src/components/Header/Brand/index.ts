export { default as headerBrandClasses } from "./brandClasses";
export * from "./brandClasses";
export { HvBrand as HvHeaderBrand } from "./Brand";
export type { HvBrandProps as HvHeaderBrandProps } from "./Brand";
