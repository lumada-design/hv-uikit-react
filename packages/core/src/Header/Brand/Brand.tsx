import { useDefaultProps } from "../../hooks/useDefaultProps";
import { HvBaseProps } from "../../types/generic";
import { HvTypography } from "../../Typography";
import { ExtractNames } from "../../utils/classes";
import { staticClasses, useClasses } from "./Brand.styles";

export { staticClasses as headerBrandClasses };

export type HvHeaderBrandClasses = ExtractNames<typeof useClasses>;

export interface HvHeaderBrandProps extends HvBaseProps {
  logo?: React.ReactNode;
  name?: string;
  classes?: HvHeaderBrandClasses;
}

/**
 * Header component is used to render a header bar with logo and brand name, navigation and actions.
 */
export const HvHeaderBrand = (props: HvHeaderBrandProps) => {
  const {
    classes: classesProp,
    logo,
    name,
    className,
    ...others
  } = useDefaultProps("HvHeaderBrand", props);

  const { classes, cx } = useClasses(classesProp);

  return (
    <div className={cx(classes.root, className)} {...others}>
      {logo}
      {logo && name && <div className={classes.separator} />}
      {name && (
        <HvTypography className={classes.brandName} variant="label">
          {name}
        </HvTypography>
      )}
    </div>
  );
};
