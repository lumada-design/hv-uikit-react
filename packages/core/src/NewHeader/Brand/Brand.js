/*
 * Copyright 2020 Hitachi Vantara Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react";
import PropTypes from "prop-types";
import withStyles from "@material-ui/core/styles/withStyles";
import HvTypography from "../../Typography";
import useUniqueId from "../../useUniqueId";

import styles from "./styles";

const Brand = ({ classes, id, logo, name }) => {
  const uniqueId = useUniqueId(id, "hv-brand-");

  return (
    <div id={uniqueId} className={classes.root}>
      {logo}
      {logo && name && <div className={classes.separator} />}
      {name && <HvTypography variant="highlightText">{name}</HvTypography>}
    </div>
  );
};

Brand.propTypes = {
  /**
   * A Jss Object used to override or extend the styles applied.
   */
  classes: PropTypes.shape({
    /**
     * Styles applied to the component root class.
     */
    root: PropTypes.string,
    /**
     * Styles applied to the separator component class.
     */
    separator: PropTypes.string
  }).isRequired,
  /**
   * Id to be applied to the root node.
   */
  id: PropTypes.string,
  /**
   * The brand image node.
   */
  logo: PropTypes.node,
  /**
   * The brand name string.
   */
  name: PropTypes.string
};

Brand.defaultProps = {
  id: undefined,
  logo: null,
  name: null
};

export default withStyles(styles, { name: "HvNewHeaderBrand" })(Brand);
