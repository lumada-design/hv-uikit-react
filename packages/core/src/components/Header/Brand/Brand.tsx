import { Typography as HvTypography } from "../../Typography";
import { BrandRoot, BrandSeparator } from "./Brand.styles";

export interface BrandProps extends BaseProps {
  logo?: React.ReactNode;
  name?: string;
}

/**
 * Header component is used to render a header bar with logo and brand name, navigation and actions.
 */
export const Brand = ({ logo, name, className }: BrandProps) => {
  return (
    <BrandRoot className={className}>
      {logo}
      {logo && name && <BrandSeparator />}
      {name && <HvTypography variant="label">{name}</HvTypography>}
    </BrandRoot>
  );
};
