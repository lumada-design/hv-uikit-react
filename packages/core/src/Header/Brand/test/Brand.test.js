import React from "react";
import { mount } from "enzyme";

import HvProvider from "../../../Provider";
import Brand from "../index";

describe("Brand withStyles", () => {
  let wrapper;

  it("should be able to render", () => {
    wrapper = mount(
      <HvProvider>
        <Brand name="test" />
      </HvProvider>
    );

    expect(wrapper.find(Brand)).toMatchSnapshot();
  });
});
