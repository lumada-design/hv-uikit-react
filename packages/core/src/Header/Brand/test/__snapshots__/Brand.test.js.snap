// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand withStyles should be able to render 1`] = `
<WithStyles(Brand)
  name="test"
>
  <Brand
    classes={
      Object {
        "root": "HvHeaderBrand-root",
        "separator": "HvHeaderBrand-separator",
      }
    }
    name="test"
  >
    <div
      className="HvHeaderBrand-root"
    >
      <WithStyles(ForwardRef)
        variant="highlightText"
      >
        <ForwardRef
          classes={
            Object {
              "3xlTitle": "HvTypography-3xlTitle",
              "4xlTitle": "HvTypography-4xlTitle",
              "5xlTitle": "HvTypography-5xlTitle",
              "disabledButtonText": "HvTypography-disabledButtonText",
              "highlightText": "HvTypography-highlightText",
              "infoText": "HvTypography-infoText",
              "inlineLink": "HvTypography-inlineLink",
              "lTitle": "HvTypography-lTitle",
              "labelText": "HvTypography-labelText",
              "mTitle": "HvTypography-mTitle",
              "normalText": "HvTypography-normalText",
              "placeholderText": "HvTypography-placeholderText",
              "root": "HvTypography-root",
              "sLink": "HvTypography-sLink",
              "sText": "HvTypography-sText",
              "sTitle": "HvTypography-sTitle",
              "selectedNavText": "HvTypography-selectedNavText",
              "selectedText": "HvTypography-selectedText",
              "vizText": "HvTypography-vizText",
              "xlTitle": "HvTypography-xlTitle",
              "xsTitle": "HvTypography-xsTitle",
              "xxlTitle": "HvTypography-xxlTitle",
              "xxsTitle": "HvTypography-xxsTitle",
            }
          }
          variant="highlightText"
        >
          <p
            className="HvTypography-root HvTypography-highlightText"
          >
            test
          </p>
        </ForwardRef>
      </WithStyles(ForwardRef)>
    </div>
  </Brand>
</WithStyles(Brand)>
`;
